synthesizer/process/src/stack/finalize_types/initialize.rs

```diff
@@ -39,46 +39,25 @@ impl<N: Network> FinalizeTypes<N> {
         // Initialize a map of registers to their types.
         let mut finalize_types = Self { inputs: IndexMap::new(), destinations: IndexMap::new() };
 
-        // Initialize a list of input futures.
-        let mut input_futures = Vec::new();
-
         // Step 1. Check the inputs are well-formed. Store the input futures.
         for input in finalize.inputs() {
             // Check the input register type.
             finalize_types.check_input(stack, input.register(), input.finalize_type())?;
-
-            // If the input is a future, add it to the list of input futures.
-            if let FinalizeType::Future(locator) = input.finalize_type() {
-                input_futures.push((input.register(), *locator));
-            }
         }
 
-        // Initialize a list of consumed futures.
-        let mut consumed_futures = Vec::new();
-
-        // Step 2. Check the commands are well-formed. Store the futures consumed by the `await` commands.
+        // Step 2. Check the commands are well-formed. Make sure all the input futures are awaited.
         for command in finalize.commands() {
             // Check the command opcode, operands, and destinations.
             finalize_types.check_command(stack, finalize, command)?;
 
-            // If the command is an `await`, add the future to the list of consumed futures.
+            // If the command is an `await`, check that the register is a future.
             if let Command::Await(await_) = command {
                 // Note: `check_command` ensures that the register is a future. This is an additional check.
-                let locator = match finalize_types.get_type(stack, await_.register())? {
-                    FinalizeType::Future(locator) => locator,
-                    FinalizeType::Plaintext(..) => bail!("Expected a future in '{await_}'"),
-                };
-                consumed_futures.push((await_.register(), locator));
+                if let FinalizeType::Plaintext(_) = finalize_types.get_type(stack, await_.register())? {
+                    bail!("Expected a future in '{await_}'")
+                }
             }
         }
-
-        // Check that the input futures are consumed in the order they are passed in.
-        ensure!(
```

Comment:
Instead of removing the check, we can instead use a hashset to check that all futures have been awaited.